app/base.go

```diff
@@ -43,6 +43,9 @@ func (b BaseApp) DeliverTx(txBytes []byte) abci.ResponseDeliverTx {
 		"path", weave.GetPath(tx))
 
 	res, err := b.handler.Deliver(ctx, b.DeliverStore(), tx)
+	if err == nil {
+		b.pending = res.Diff
```

Comment:
How about `b.AddValChange(res.Diff)` ?

Maybe you have two transactions in the same block that change the set?

Reply:
@ethanfrey Thanks, I knew I missed something cause it didn't look right.

Reply:
Actually, what about this: https://github.com/confio/weave/blob/d22d3298750aff98d0bc22ffb5dd407d6f1e2943/app/base.go#L84 ? 

I'm going to investigate this whole thing more to understand when all the conditions are met.
We definitely don't need this to happen in two places.

Reply:
@ethanfrey On the other hand I can see that ticker implementation does not exist, so this shouldn't be a concern.

Also, I'm not sure what happens when we get duplicate entries in the Diff, but as long as they have the same values - it shouldn't matter much.